earth_enterprise/src/fusion/autoingest/sysman/QuadKeyTree_unittest.cpp

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2019 The Open GEE Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+#include <autoingest/sysman/InsetTilespaceIndex.h>
+#include "common/quadtreepath.h"
+
+using namespace std;
+
+class QuadKeyTreeTest : public testing::Test {
+ public:
+  QuadKeyTreeTest()
+  {
+  }
+};
+
+// TEST DATA
+// std::array<std::string, 9> data {{ "this", "is", "some", "string", "data", 
+//                                    "one fish", "two fish" , "red fish", "blue fish"}};
+QuadtreePath qtp01("01");
+QuadtreePath qtp011("011");
+QuadtreePath qtp012("012");
+QuadtreePath qtp0123("0123");
+QuadtreePath qtp1("1");
```

Comment:
Does QTP 1 even make sense? I thought QTPs always started with 0.

Reply:
My understanding is that level 0 is implied but not represented in these objects. That's the sense I got while looking through the code in quadtreepath_unittest.cpp.
i.e., the first digit in the paths shown here represent the tile at level 1.